patches/tModLoader/Terraria/ModLoader/AccessorySlotLoader.cs

```diff
@@ -0,0 +1,536 @@
+				SoundEngine.PlaySound(12);
+			}
+
+			int num45 = ((!ModSlotPlayer.scrollSlots) ? 0 : 1);
+			Main.HoverItem = new Item();
+			Main.hoverItemName = scrollStackLang[num45];
+		}
+
+		// This is a hacky solution to make it very vanilla-esque, at the cost of not actually using a UI proper. 
+		internal void DrawScrollbar(int num20, int skip) {
+			int xLoc = Main.screenWidth - 64 - 28;
+
+			int accessoryPerColumn = GetAccessorySlotPerColumn(num20);
+			int slotsToRender = list.Count + MaxVanillaSlotCount - skip;
+			int scrollIncrement = slotsToRender - accessoryPerColumn;
+
+			if (scrollIncrement < 0) {
+				accessoryPerColumn = slotsToRender;
+				scrollIncrement = 0;
+			}
+
+			int chkMax = (int)((float)(num20) + (float)(((accessoryPerColumn) + 3) * 56) * Main.inventoryScale) + 4;
+			int chkMin = (int)((float)(num20) + (float)((0 + 3) * 56) * Main.inventoryScale) + 4;
+
+			UIScrollbar scrollbar = new UIScrollbar();
+
+			Rectangle rectangle = new Rectangle(xLoc + 47 + 6, chkMin, 5, chkMax - chkMin);
+			scrollbar.DrawBar(Main.spriteBatch, Main.Assets.Request<Texture2D>("Images/UI/Scrollbar").Value, rectangle, Color.White);
+			
+			int barSize = (chkMax - chkMin) / (scrollIncrement + 1);
+			rectangle = new Rectangle(xLoc + 47 + 5, chkMin + ModSlotPlayer.scrollbarSlotPosition * barSize, 3, barSize);
+			scrollbar.DrawBar(Main.spriteBatch, Main.Assets.Request<Texture2D>("Images/UI/ScrollbarInner").Value, rectangle, Color.White);
+
+			rectangle = new Rectangle(xLoc - 47 * 2, chkMin, 47 * 3, chkMax - chkMin);
+			if (!(rectangle.Contains(new Point(Main.mouseX, Main.mouseY)) && !PlayerInput.IgnoreMouseInterface)) {
+				return;
+			}
+
+			PlayerInput.LockVanillaMouseScroll("ModLoader/Acc");
+
+			int scrollDelta = ModSlotPlayer.scrollbarSlotPosition + (int)PlayerInput.ScrollWheelDelta / 120;
+			scrollDelta = Math.Min(scrollDelta, scrollIncrement);
+			scrollDelta = Math.Max(scrollDelta, 0);
+
+			ModSlotPlayer.scrollbarSlotPosition = scrollDelta;
+			PlayerInput.ScrollWheelDelta = 0;
+		}		
+
+		/// <summary>
+		/// Draws Vanilla and Modded Accessory Slots
+		/// </summary>
+		public bool Draw(int num20, int skip, bool modded, int slot, Color color) {
+			bool flag3;
+			bool flag4 = false;
+
+			if (modded) {
+				flag3 = !ModdedIsAValidEquipmentSlotForIteration(slot);
+				flag4 = !ModdedCanSlotBeShown(slot);
+			}
+			else {
+				flag3 = !Player.IsAValidEquipmentSlotForIteration(slot);
+				if (slot == 8) {
+					flag4 = (slot == 8) && !Player.CanDemonHeartAccessoryBeShown();
+				}
+				else if (slot == 9) {
+					flag4 = !Player.CanMasterModeAccessoryBeShown();
+				}
+			}
+
+			if (flag4 && flag3) {
+				return false;
+			}
+
+			Main.inventoryBack = flag3 ? new Color(80, 80, 80, 80) : color;
+
+			int yLoc = 0, xLoc = 0;
+			bool customLoc = false;
+
+			if (modded) {
+				ModAccessorySlot mAccSlot = list[slot];
+				customLoc = mAccSlot.XLoc != 0 || mAccSlot.YLoc != 0;
```

Comment:
I don't suppose there's any actual desire for custom location (0, 0)? Might be better to return -1 for 'no custom location'? Not a big deal either way.